src/libs/sso-utils.js

```diff
@@ -285,16 +226,11 @@ export const checkEmailExists = async (credentials, userInfo) => {
   }
 };
 
-export const updateUser = async (credentials, userInfo) => {
-  checkCredentialValid(credentials);
+export const updateUser = async userInfo => {
   try {
-    const options = {
-      headers: {
-        'Content-Type': SSO_REQUEST.CONTENT_TYPE_JSON,
-        Authorization: `Bearer ${credentials.token}`,
-      },
-      uri: url.resolve(credentials.uri, `${SSO_SUB_URI.USER}/${userInfo.id}`),
-      method: 'PUT',
+    let options = await setRequestHeader(`${SSO_SUB_URI.USER}/${userInfo.id}`, 'PUT');
+    options = {
+      ...options,
       body: {
```

Comment:
the one interesting thing I find here is you use `setRequestHeader` to provide some partial configuration to the request object and then further configure it later down. It almost seems like `setRequestHeader` is a `RequestBuilder` in which it should be able to set/get things like headers, body, and other properties. Just my thoughts

Reply:
renamed to requestBuilder